.github/workflows/50_go_build-one-arch-one-os-one-path.yml

```diff
@@ -179,27 +217,39 @@ jobs:
             ${{ steps.setup-keys.outputs.go-build }}
           key: ${{ steps.setup-keys.outputs.go-build-cache-key-cmd }}
       - name: Go Build
+        if: ${{ inputs.run-build || (inputs.retag-base-image && steps.docker-retag.outcome != 'success') }}
+        env:
+          GOARCH: ${{ steps.setup-keys.outputs.goarch }}
+          GOOS: ${{ steps.setup-keys.outputs.goos }}
         run: |
           GOARCH=${{ steps.setup-keys.outputs.goarch }} \
           GOOS=${{ steps.setup-keys.outputs.goos }} \
           go build -v -o "build/$(basename ${{ inputs.cmd-path }})" ${{ inputs.cmd-path }}
       - name: Pack as Docker image
-        if: ${{ inputs.pack-to-docker-image }}
+        if: ${{ inputs.pack-to-docker-image || (inputs.retag-base-image && steps.docker-retag.outcome != 'success') }}
```

Comment:
`steps.docker-retag.outcome != 'success'` wouldn't this need `docker-retag` to be in the `needs` list?